provider/machine_pool_resource.go

```diff
@@ -455,6 +560,71 @@ func (r *MachinePoolResource) Update(ctx context.Context, request tfsdk.UpdateRe
 	response.Diagnostics.Append(diags...)
 }
 
+// Validate the machine pool's settings that pertain to availability zones.
+// Returns whether the machine pool is/will be multi-AZ.
+func (r *MachinePoolResource) validateAZConfig(state *MachinePoolState) (bool, error) {
+	resp, err := r.collection.Cluster(state.Cluster.Value).Get().Send()
+	if err != nil {
+		return false, fmt.Errorf("failed to get information for cluster %s: %v", state.Cluster.Value, err)
+	}
+	cluster := resp.Body()
+	isMultiAZCluster := cluster.MultiAZ()
+	clusterAZs := cluster.Nodes().AvailabilityZones()
+	clusterSubnets := cluster.AWS().SubnetIDs()
+
+	poolAZs, err := common.StringListToArray(state.AvailabilityZones)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse availability_zones: %v", err)
+	}
+	poolSubnets, err := common.StringListToArray(state.SubnetIDs)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse subnet_ids: %v", err)
+	}
+
+	// Can be unspecified, single entry(1az pool), or match the cluster's AZ count
+	if len(poolAZs) != 0 && len(poolAZs) != 1 && len(poolAZs) != len(clusterAZs) {
+		return false, fmt.Errorf("Invalid number of entries for \"availability_zones\"")
+	}
+	if len(poolSubnets) != 0 && len(poolSubnets) != 1 && len(poolSubnets) != len(clusterSubnets) {
+		return false, fmt.Errorf("Invalid number of entries for \"subnet_ids\"")
+	}
+
+	// Any specified AZs must be in the cluster's AZs
+	if len(poolAZs) > 0 {
+		for _, poolAZ := range poolAZs {
+			poolInClusterAZ := false
+			for _, clusterAZ := range clusterAZs {
+				if clusterAZ == poolAZ {
+					poolInClusterAZ = true
+				}
+			}
+			if !poolInClusterAZ {
+				return false, fmt.Errorf("Setting \"availability_zone\" is not compatible with the cluster's AZs")
+			}
+		}
+	}
+
+	if len(poolSubnets) > 0 {
+		for _, poolSubnet := range poolSubnets {
+			poolInClusterSubnet := false
+			for _, clusterSubnet := range clusterSubnets {
+				if clusterSubnet == poolSubnet {
+					poolInClusterSubnet = true
+				}
+			}
+			if !poolInClusterSubnet {
+				return false, fmt.Errorf("Setting \"subnet_ids\" is not compatible with the cluster's subnets")
+			}
+		}
+	}
+
+	// It's a multi-az pool if the cluster is multi-az and they didn't tell us
+	// which az to put the pool into
+	isMultiAZPool := isMultiAZCluster && len(poolAZs) != 1 && len(poolSubnets) != 1
```

Comment:
As I see in rosa cli - you can't see AZs and submets together in one machine pool  - I think it should be changed to :
```suggestion
	isMultiAZPool := isMultiAZCluster && (len(poolAZs) > 1 || len(poolSubnets) > 1)
```

Reply:
It is not necessary to specify subnets or pool AZs in the case where you want the same "multi"-ness as the cluster.
In that case, len of both of those could be 0. For a multiAZ cluster, that would lead us to mistakenly think it's a single AZ pool, when it's not.